### PR TITLE
DM-27387: Implement scattering magnitudes for ApFakes

### PR DESCRIPTION
**Pull request by morriscb**

### CODE REVIEW

File: python/lsst/pipe/tasks/matchFakes.py [hunk 1]
```diff
@@ -31,7 +31,9 @@
 
 __all__ = ["MatchFakesTask",
            "MatchFakesConfig",
-           "MatchFakesConnections"]
+           "MatchFakesConnections",
+           "MatchVariableFakesConfig",
+           "MatchVariableFakesTask"]
```
Comment (jmeyers314) [inline python/lsst/pipe/tasks/matchFakes.py:36]: Is there a reason to include `MatchFakesConnections` here but not `MatchVariableFakesConnections`?

File: python/lsst/pipe/tasks/matchFakes.py [hunk 2]
```diff
@@ -193,3 +199,95 @@ def _getVectors(self, ras, decs):
         vectors[:, 1] = np.cos(decs) * np.sin(ras)
 
         return vectors
+
+
+class MatchVariableFakesConnections(MatchFakesConnections):
+    ccdVisitFakeMagnitudes = connTypes.Input(
+        doc="Catalog of fakes with magnitudes scatted for this ccdVisit.",
```
Comment (jmeyers314) [inline python/lsst/pipe/tasks/matchFakes.py:206]: scatted -> scattered.

File: python/lsst/pipe/tasks/processCcdWithFakes.py [hunk 3]
```diff
@@ -465,3 +468,140 @@ def copyCalibrationFields(self, calibCat, sourceCat, fieldsToCopy):
                 calibSrc.setFootprint(calibSrcFootprint)
 
         return newCat
+
+
+class ProcessCcdWithVariableFakesConnections(ProcessCcdWithFakesConnections):
+    ccdVisitFakeMagnitudes = cT.Output(
+        doc="Catalog of fakes with magnitudes scatted for this ccdVisit.",
```
Comment (jmeyers314) [inline python/lsst/pipe/tasks/processCcdWithFakes.py:475]: scatted -> scattered

File: python/lsst/pipe/tasks/processCcdWithFakes.py [hunk 4]
```diff
@@ -465,3 +468,140 @@ def copyCalibrationFields(self, calibCat, sourceCat, fieldsToCopy):
...
+                    Catalog to take the information about which sources were used for calibration from.
+        sfdSourceCat : `lsst.afw.table.SourceCatalog`
+                    Default : None
+                    Catalog produced by singleFrameDriver, needed to copy some calibration flags from.
+
+        Returns
+        -------
+        resultStruct : `lsst.pipe.base.struct.Struct`
+            contains : outputExposure : `lsst.afw.image.exposure.exposure.ExposureF`
+                       outputCat : `lsst.afw.table.source.source.SourceCatalog`
```
Comment (jmeyers314) [inline python/lsst/pipe/tasks/processCcdWithFakes.py:530]: missing `ccdVisitFakeMagnitudes`

**Review by jmeyers314 — APPROVED** [review]
LGTM.  A few minor comments/questions.

File: python/lsst/pipe/tasks/processCcdWithFakes.py [hunk 5]
```diff
@@ -465,3 +468,140 @@ def copyCalibrationFields(self, calibCat, sourceCat, fieldsToCopy):
...
+
+        Returns
+        -------
+        dataFrame : `pandas.DataFrame`
+            DataFrame containing the values of the magnitudes to that will
+            be inserted into this ccdVisit.
+        """
+        expId = exposureIdInfo.expId
+        rng = np.random.default_rng(expId)
+        magScatter = np.zeros(len(fakeCat))
```
[the remark this answers is not in the document]
Comment (morriscb) [reply python/lsst/pipe/tasks/processCcdWithFakes.py:601]: Nice catch. Was a left over from a previous verions.

File: python/lsst/pipe/tasks/matchFakes.py (hunk 1 above)
Comment (morriscb) [reply python/lsst/pipe/tasks/matchFakes.py:36]: Not really, I'll remove the previous Connections.

**Pull request merged**